src/test/java/marquez/db/DatasetDaoTest.java

```diff
@@ -77,13 +100,36 @@ public void testInsert() {
     assertThat(rowsAfter).isEqualTo(rowsBefore + 1);
   }
 
+  @Test(expected = UnableToExecuteStatementException.class)
+  public void testIdempotentInsert() {
+    final int rowsBefore = datasetDao.count();
+
+    final DatasetRow newDatasetRow =
+        newDatasetRowWith(namespaceRow.getUuid(), datasourceRow.getUuid());
+    datasetDao.insert(newDatasetRow);
+
+    final int rowsAfter = datasetDao.count();
+    assertThat(rowsAfter).isEqualTo(rowsBefore + 1);
+
+    datasetDao.insert(newDatasetRow);
```

Comment:
```java
assertThatExceptionOfType(UnableToExecuteStatementException.class)
    .isThrownBy(() -> datasetDao.insert(newDatasetRow));
```

Reply:
Nice - I like this approach more b/c it makes the test more precise by indicating just where it's supposed to throw the exception.